src/main/java/com/synopsys/integration/alert/provider/blackduck/BlackDuckValidator.java

```diff
@@ -0,0 +1,93 @@
+                final String errorMessage = "BlackDuck Provider invalid: URL missing";
+                systemMessageUtility.addSystemMessage(errorMessage, SystemMessageSeverity.WARNING, SystemMessageType.BLACKDUCK_PROVIDER_URL_MISSING);
+            } else {
+                final String blackDuckUrlString = blackDuckUrlOptional.get();
+                final Boolean trustCertificate = BooleanUtils.toBoolean(alertProperties.getAlertTrustCertificate().orElse(false));
+                final Integer timeout = blackDuckProperties.getBlackDuckTimeout();
+                logger.debug("  -> BlackDuck Provider URL found validating: {}", blackDuckUrlString);
+                logger.debug("  -> BlackDuck Provider Trust Cert: {}", trustCertificate);
+                logger.debug("  -> BlackDuck Provider Timeout: {}", timeout);
+                final URL blackDuckUrl = new URL(blackDuckUrlString);
+                if ("localhost".equals(blackDuckUrl.getHost())) {
+                    logger.warn("  -> BlackDuck Provider Using localhost...");
+                    final String blackDuckWebServerHost = blackDuckProperties.getPublicBlackDuckWebserverHost().orElse("");
+                    logger.warn("  -> BlackDuck Provider Using localhost because PUBLIC_BLACKDUCK_WEBSERVER_HOST environment variable is set to {}", blackDuckWebServerHost);
+                    systemMessageUtility.addSystemMessage("BlackDuck Provider Using localhost", SystemMessageSeverity.WARNING, SystemMessageType.BLACKDUCK_PROVIDER_LOCALHOST);
+                }
+                final IntLogger intLogger = new Slf4jIntLogger(logger);
+                final Optional<BlackDuckServerConfig> blackDuckServerConfig = blackDuckProperties.createBlackDuckServerConfig(intLogger);
+                if (blackDuckServerConfig.isPresent()) {
+                    final Boolean canConnect = blackDuckServerConfig.get().canConnect(intLogger);
+                    if (canConnect) {
+                        logger.info("  -> BlackDuck Provider Valid!");
+                    } else {
+                        final String message = "Can not connect to the BlackDuck server with the current configuration.";
+                        connectivityWarning(systemMessageUtility, message);
+                    }
+                } else {
+                    final String message = "The BlackDuck configuration is not valid.";
+                    connectivityWarning(systemMessageUtility, message);
+                }
+            }
+        } catch (final MalformedURLException | IntegrationException | AlertRuntimeException ex) {
+            logger.error("  -> BlackDuck Provider Invalid; cause: {}", ex.getMessage());
+            logger.debug("  -> BlackDuck Provider Stack Trace: ", ex);
+            systemMessageUtility.addSystemMessage("BlackDuck Provider invalid: " + ex.getMessage(), SystemMessageSeverity.WARNING, SystemMessageType.BLACKDUCK_PROVIDER_CONNECTIVITY);
```

Comment:
Do we want to show the exception message in system messages? I think, given the limited screen real estate, we should omit it since we already divulge that information when a "Test Connection" is performed.

Reply:
In this version of Alert we don't need to show the exception message anymore.  With the 3.x versions we were required to have a valid Black Duck connection to even log in. That error message would show the certificate errors for example.  If the provider is invalid they can now test the connection so we don't need that detail anymore in the UI.